internal/mode/static/nginx/runtime/verify_test.go

```diff
@@ -68,7 +89,88 @@ func TestVerifyClient(t *testing.T) {
 		t.Run(test.name, func(t *testing.T) {
 			g := NewWithT(t)
 
-			err := c.waitForCorrectVersion(test.ctx, test.expectedVersion)
+			err := c.waitForCorrectVersion(test.ctx, test.expectedVersion, "/childfile", []byte("1 2 3"), test.readFile)
+
+			if test.expectError {
+				g.Expect(err).To(HaveOccurred())
+			} else {
+				g.Expect(err).ToNot(HaveOccurred())
+			}
+		})
+	}
+}
+
+func TestEnsureNewNginxWorkers(t *testing.T) {
+	previousContents := []byte("1 2 3")
+	newContents := []byte("4 5 6")
+
+	readFileError := func(string) ([]byte, error) {
+		return nil, errors.New("error")
+	}
+
+	readFilePrevious := func(string) ([]byte, error) {
+		return previousContents, nil
+	}
+
+	readFileNew := func(string) ([]byte, error) {
+		return newContents, nil
+	}
+
+	ctx := context.Background()
+
+	cancellingCtx, cancel := context.WithCancel(ctx)
+	time.AfterFunc(10000*time.Millisecond, cancel)
```

Comment:
Does it have to be 10s?

on my machine, running all tests takes 17 seconds, so 10s is more than a half of it.

Reply:
Removed a couple of zeroes 😁